### PR TITLE
Support proxy with authentication

### PR DESCRIPTION
**Pull request by ghost**
Usage:
```
from instapy import InstaPy 
from proxy_extension import create_proxy_extension

insta_username = ''
insta_password = ''
proxy = 'login:password@123.123.123.123:8000'
proxy_chrome_extension = create_proxy_extension(proxy)

session = InstaPy(
    username=insta_username,
    password=insta_password,
    proxy_chrome_extension=proxy_chrome_extension,
    headless_browser=False,
    multi_logs=True,
)

```

### CODE REVIEW

**Comment by sionking** [comment]
Is this my solution ? 
https://github.com/timgrossmann/InstaPy/issues/1472
Thanks,

**Comment by sionking** [comment]
Thank you for merging (I don't have time soory)

File: instapy/instapy.py [hunk 1]
```diff
@@ -1158,6 +1163,9 @@ def interact_by_users(self,
             # Reset like counter for every username
             liked_img = 0
 
+            # Will we follow this user?
+            following = random.randint(0, 100) <= self.follow_percentage
```
Comment (timgrossmann) [inline instapy/instapy.py:1167]: Why are you pulling this out of the for loop?

**Review by timgrossmann — COMMENTED** [review]
Please also add some documentation in the README file

File: instapy/instapy.py (hunk 1 above)
Comment (ghost) [reply instapy/instapy.py:1167]: this is another PR. Description is here https://github.com/timgrossmann/InstaPy/pull/1535

Comment (ghost) [reply instapy/instapy.py:1167]: I have added some documentation in the README file about proxy with authentication.

Comment (timgrossmann) [reply instapy/instapy.py:1167]: You can't leave that here, please delete that line so we can introduce the proxy feature by itself and merge the other PR separately

**Comment by SeanFmcdonnell** [comment]
Adding a proxy was not working with my account I still got a `Cookie file not found, creating cookie...` and `Wrong login data` error...

**Comment by timgrossmann** [comment]
@SeanFmcdonnell It's not an error, it's just a note that it's creating a new cookie since it could not find a saved one.

**Comment by SeanFmcdonnell** [comment]
After the message that it creates the cookie, it ends the session. I’m not
sure if it’s saving. I’ll login on a local browser and will see a
suspicious login attempt.

**Comment by timgrossmann** [comment]
@SeanFmcdonnell Yeah you have to get around that using the suspicious login "feature" once. After that you can run InstaPy more easily.
Or you take the approach of @sionking by simply doing the login manually instead of automated.

**Pull request merged**